src/plugins/dashboard/public/application/top_nav/editor_menu.tsx

```diff
@@ -0,0 +1,210 @@
+    usageCollection,
+    DashboardConstants.DASHBOARDS_ID
+  );
+
+  const createNewAggsBasedVis = useCallback(
+    async () =>
+      visualizations.showNewVisModal({
+        originatingApp: DashboardConstants.DASHBOARDS_ID,
+        outsideVisualizeApp: true,
+        showAggsSelection: true,
+      }),
+    [visualizations]
+  );
+
+  const getVisTypesByGroup = (group: VisGroups) =>
+    visualizations
+      .getByGroup(group)
+      .sort(({ name: a }: BaseVisType | VisTypeAlias, { name: b }: BaseVisType | VisTypeAlias) => {
+        if (a < b) {
+          return -1;
+        }
+        if (a > b) {
+          return 1;
+        }
+        return 0;
+      })
+      .filter(({ hidden }: BaseVisType) => !hidden);
+
+  const promotedVisTypes = getVisTypesByGroup(VisGroups.PROMOTED);
+  const visTypeAliases = visualizations
+    .getAliases()
+    .sort(({ promotion: a = false }: VisTypeAlias, { promotion: b = false }: VisTypeAlias) =>
+      a === b ? 0 : a ? -1 : 1
+    );
+
+  const factories = embeddable
+    ? Array.from(embeddable.getEmbeddableFactories()).filter(
+        ({ type, isEditable, canCreateNew, isContainerType }) =>
+          isEditable() && !isContainerType && canCreateNew() && type !== 'visualization'
+      )
+    : [];
+
+  const factoryGroupMap: Record<string, FactoryGroup> = {};
+  const ungroupedFactories: EmbeddableFactoryDefinition[] = [];
+  let panelCount = 1;
+
+  factories.forEach((factory: EmbeddableFactoryDefinition, index) => {
+    const { grouping } = factory;
+
+    if (grouping) {
+      grouping.forEach((group) => {
+        if (factoryGroupMap[group.id]) {
+          factoryGroupMap[group.id].factories.push(factory);
+        } else {
+          factoryGroupMap[group.id] = {
+            id: group.id,
+            appName: group.getDisplayName ? group.getDisplayName({ embeddable }) : group.id,
+            icon: (group.getIconType ? group.getIconType({ embeddable }) : 'empty') as IconType,
+            factories: [factory],
+            panelId: panelCount,
+          };
+
+          panelCount++;
+        }
+      });
+    } else {
+      ungroupedFactories.push(factory);
+    }
+  });
+
+  const getVisTypeMenuItem = (visType: BaseVisType): EuiContextMenuPanelItemDescriptor => {
+    const { name, title, titleInWizard, description, icon = 'empty' } = visType;
+
+    return {
+      name: titleInWizard || title,
+      icon: icon as string,
+      onClick: createNewVisType(visType),
+      'data-test-subj': `visType-${name}`,
+      toolTipContent: description,
+    };
+  };
+
+  const getVisTypeAliasMenuItem = (
+    visTypeAlias: VisTypeAlias
+  ): EuiContextMenuPanelItemDescriptor => {
+    const { name, title, description, icon = 'empty' } = visTypeAlias;
+
+    return {
+      name: title,
+      icon,
+      onClick: createNewVisType(visTypeAlias),
+      'data-test-subj': `visType-${name}`,
+      toolTipContent: description,
+    };
+  };
+
+  const getEmbeddableFactoryMenuItem = (
```

Comment:
The embeddable menu item is missing a tooltip, can you add one to indicate what action will happen on click?

Reply:
Sure, but I think I'll need to reach out to the ML and Logs teams for the copy, since I don't think we should use a generic tooltip here.